packages/merge-styles/src/styleToClassName.ts

```diff
@@ -141,33 +141,86 @@ export function serializeRuleEntries(ruleEntries: { [key: string]: string | numb
   return allEntries.join('');
 }
 
-export function styleToClassName(...args: IStyle[]): string {
+export interface IRegistration {
+  className: string;
+  key: string;
+  args: IStyle[];
+  rulesToInsert: string[];
+}
+
+export function styleToRegistration(...args: IStyle[]): IRegistration | undefined {
   const rules: IRuleSet = extractRules(args);
   const key = getKeyForRules(rules);
-  let className = '';
 
   if (key) {
     const stylesheet = Stylesheet.getInstance();
+    const registration: Partial<IRegistration> = {
+      className: stylesheet.classNameFromKey(key),
+      key,
+      args
+    };
+
+    if (!registration.className) {
+      registration.className = stylesheet.getClassName(getDisplayName(rules));
+      const rulesToInsert: string[] = [];
+
+      for (const selector of rules.__order) {
+        rulesToInsert.push(
+          selector,
+          serializeRuleEntries(rules[selector])
+        );
+      }
+      registration.rulesToInsert = rulesToInsert;
+    }
 
-    className = stylesheet.classNameFromKey(key)!;
-
-    if (!className) {
-      className = stylesheet.getClassName(getDisplayName(rules));
-      const registeredRules: string[] = [];
+    return registration as IRegistration;
+  }
+}
 
-      for (let selector of rules.__order) {
-        const rulesToInsert: string = serializeRuleEntries(rules[selector]);
+export function applyRegistration(
+  registration: IRegistration,
+  classMap?: { [key: string]: string }
+): void {
+  const stylesheet = Stylesheet.getInstance();
+  const { className, key, args, rulesToInsert } = registration;
+
+  if (rulesToInsert) {
+    for (let i = 0; i < rulesToInsert.length; i += 2) {
```

Comment:
The reasons for the weird indexing and non-standard looping is not at all obvious. If it's intended, can you add comments to explain what's going on?

Reply:
Added comment here.